libs/ml/src/layers/convolution_1d.cpp

```diff
@@ -0,0 +1,131 @@
+  : kernel_size_{kernel_size}
+  , input_channels_{input_channels}
+  , output_channels_{output_channels}
+  , stride_size_{stride_size}
+{
+  std::string input =
+      this->template AddNode<fetch::ml::ops::PlaceHolder<TensorType>>(name + "_Input", {});
+
+  std::string weights =
+      this->template AddNode<fetch::ml::ops::Weights<TensorType>>(name + "_Weights", {});
+
+  TensorType weights_data(
+      std::vector<SizeType>{{output_channels_, input_channels_, kernel_size_, 1}});
+  fetch::ml::ops::Weights<TensorType>::Initialise(weights_data, 1, 1, init_mode, seed);
+  this->SetInput(weights, weights_data);
+
+  std::string output = this->template AddNode<fetch::ml::ops::Convolution1D<TensorType>>(
+      name + "_Conv1D", {input, weights}, stride_size_);
+
+  output = fetch::ml::details::AddActivationNode<TensorType>(activation_type, this,
+                                                             name + "_Activation", output);
+
+  this->AddInputNode(input);
+  this->SetOutputNode(output);
+
+  this->Compile();
+}
+
+template <typename TensorType>
+void Convolution1D<TensorType>::SetOpSaveableParams(SPType const &sp)
+{
+  // assign layer specific params
+  kernel_size_     = sp.kernel_size;
+  input_channels_  = sp.input_channels;
+  output_channels_ = sp.output_channels;
+  stride_size_     = sp.stride_size;
+}
+
+template <class TensorType>
+std::shared_ptr<OpsSaveableParams> Convolution1D<TensorType>::GetOpSaveableParams()
+{
+  // get all base classes saveable params
+  std::shared_ptr<OpsSaveableParams> sgsp = SubGraph<TensorType>::GetOpSaveableParams();
+
+  auto ret = std::make_shared<SPType>();
+
+  // copy subgraph saveable params over
+  auto sg_ptr1 = std::dynamic_pointer_cast<typename SubGraph<TensorType>::SPType>(sgsp);
+  auto sg_ptr2 = std::dynamic_pointer_cast<typename SubGraph<TensorType>::SPType>(ret);
+  *sg_ptr2     = *sg_ptr1;
+
+  // asign layer specific params
+  ret->kernel_size     = kernel_size_;
+  ret->input_channels  = input_channels_;
+  ret->output_channels = output_channels_;
+  ret->stride_size     = stride_size_;
+
+  return ret;
+}
+
+template <class TensorType>
+std::vector<fetch::math::SizeType> Convolution1D<TensorType>::ComputeOutputShape(
+    VecTensorType const &inputs) const
+{
+  TensorType weights_data(
+      std::vector<SizeType>{{output_channels_, input_channels_, kernel_size_, 1}});
+  return fetch::ml::ops::Convolution1D<TensorType>(stride_size_)
+      .ComputeOutputShape({inputs.at(0), std::make_shared<TensorType>(weights_data)});
+}
+
+///////////////////////////////
+/// EXPLICIT INSTANTIATIONS ///
+///////////////////////////////
+
+// template class Convolution1D<math::Tensor<int8_t>>;
+// template class Convolution1D<math::Tensor<int16_t>>;
+template class Convolution1D<math::Tensor<int32_t>>;
+template class Convolution1D<math::Tensor<int64_t>>;
+// template class Convolution1D<math::Tensor<uint8_t>>;
+// template class Convolution1D<math::Tensor<uint16_t>>;
```

Comment:
are these commented instantiations not needed?

Reply:
they don't work for now, but should work in future